realm/realm-library/src/main/java/io/realm/SetValueOperator.java

```diff
@@ -1069,6 +1077,116 @@ boolean retainAllInternal(Collection<?> c) {
     }
 }
 
+/**
+ * TODO
+ */
+class RealmModelSetOperator<T extends RealmModel> extends SetValueOperator<T> {
+
+    public RealmModelSetOperator(BaseRealm baseRealm, OsSet osSet, Class<T> valueClass) {
+        super(baseRealm, osSet, valueClass);
+    }
+
+    @Override
+    boolean add(@Nullable T value) {
+        // RealmModel sets doesnt allow null values
+        if (value == null) {
+            throw new IllegalArgumentException("Null values cannot be added into this set.");
```

Comment:
This should throw a `NullPointerException` according to the Java `Set` interface: https://docs.oracle.com/javase/7/docs/api/java/util/Set.html#add(E)

Reply:
You should in general double-check the `Set` interface for restrictions in parameters and how the should throw. I completely ignored these for dictionaries and had to change them last week 🙊